### DISCUSSION TITLE
(session.sleep, session.wakeup) or (session.dump, session.load)

### DISCUSSION THREAD

**Discussion by Cirn09** [FlareSolverr/FlareSolverr#752] [Ideas]
Sometimes I need to use the same session for a long time to complete a task, but currently the browser process keeps running after the session is created and before it is destroyed. This may not be friendly for devices with small memory.

Perhaps a `session.sleep` command could be added to hibernate the target session, and `session.wakeup` command to bring the target session back up.
Or more simply `session.dump`, which prints all the state of the current session; and `session.load`, which resumes from the previous state.

**Comment by ngosang** [comment]
This will make the FlareSolverr code more complex and I think it will cause even more complex issues in the future. Closing the Chrome browser properly, keeping all the directories with the session data, and restoring Chrome again is more difficult than it seems. Communication with Chrome is done through the Selenium API which has many limitations.

Use case 1:
* Few queries
* Do not use sessions

Use case 2:
* Low resources, many queries in a short period of time and then idle for a while
* Use sessions but close them after you finish the requests.

Use case 3:
* Low resources, many queries
* Use sessions

**Reply by Cirn09** [reply]
Perhaps a simple `session.dump` and `session.load` that only exports all cookies could be implemented first?
Currently the `request.get`'s response seems to contain only cookies used in this request.
